disttask/framework/dispatcher/dispatcher.go

```diff
@@ -476,13 +478,23 @@ func (d *dispatcher) GetAllSchedulerIDs(ctx context.Context, gTaskID int64) ([]s
 	}
 	ids := make([]string, 0, len(schedulerIDs))
 	for _, id := range schedulerIDs {
-		if _, ok := serverInfos[id]; ok {
+		if ok := matchServerInfo(serverInfos, id); ok {
 			ids = append(ids, id)
 		}
 	}
 	return ids, nil
 }
 
+func matchServerInfo(serverInfos map[string]*infosync.ServerInfo, schedulerID string) bool {
+	for _, serverInfo := range serverInfos {
+		serverID := disttaskutil.GenerateExecID(serverInfo.IP, serverInfo.Port)
```

Comment:
Can we store all serverID in some place, and avoid call GenerateExecID in every time we call 'matchServerInfo'?

Reply:
For now, this infor is stored in PD, and from TiDB side we do not plan to store it in TiDB local, because for a lease period, PD may update the newest TiDB server info. later we will implement dynamic subtast schedule, it need above logic to get real time information.